sklearn/preprocessing/tests/test_data.py

```diff
@@ -2224,3 +2225,218 @@ def test_quantile_transform_valid_axis():
+    assert_equal(X_multi_t.shape, (size, n_values * (n_values - 1)))
+
+    expected = np.hstack([encoder.fit_transform(X_multi[:, i:(i + 1)])
+                          for i in range(X_multi.shape[1])])
+    assert_array_equal(expected, X_multi_t)
+
+
+def test_unary_encoder_dense_sparse():
+    # test dense output in comparison to sparse results.
+    n_values = np.random.randint(1, 10)
+    size = np.random.randint(1, 10)
+
+    sparse_encoder = UnaryEncoder(n_values, sparse=True)
+    dense_encoder = UnaryEncoder(n_values)
+
+    X = _generate_random_features_matrix(n_values, size)
+    X_trans_sparse = sparse_encoder.fit_transform(X)
+    X_trans_dense = dense_encoder.fit_transform(X)
+
+    assert_array_equal(X_trans_sparse.toarray(), X_trans_dense)
+
+
+def test_unary_encoder_handle_greater():
+    X = np.array([[0, 2, 1], [1, 0, 3], [2, 0, 2]])
+    y = np.array([[4, 1, 1]])
+
+    # Test that encoder raises error for greater features during transform.
+    encoder = UnaryEncoder(handle_greater='error')
+    encoder.fit(X)
+    assert_raises(ValueError, encoder.transform, y)
+
+    encoder = UnaryEncoder(handle_greater='error')
+    assert_array_equal(encoder.fit_transform(y),
+                       np.array([[1.,  1.,  1.,  1.,  1.,  1.]]))
+
+    # Test that encoder raises error for greater features during fit when
+    # n_values is explicitly set.
+    encoder = UnaryEncoder(handle_greater='error', n_values=[2, 3, 4])
+    assert_raises(ValueError, encoder.fit, X)
+
+    encoder = UnaryEncoder(handle_greater='error', n_values=[2, 3, 4])
+    assert_raises(ValueError, encoder.fit_transform, X)
+
+    encoder = UnaryEncoder(handle_greater='error', n_values=[5, 2, 2])
+    encoder.fit(y)
+    assert_array_equal(encoder.transform(y),
+                       np.array([[1.,  1.,  1.,  1.,  1.,  1.]]))
+
+    encoder = UnaryEncoder(handle_greater='error', n_values=[5, 2, 2])
+    assert_array_equal(encoder.fit_transform(y),
+                       np.array([[1.,  1.,  1.,  1.,  1.,  1.]]))
+
+    # Test the clip option.
+    encoder = UnaryEncoder(handle_greater='clip')
+    encoder.fit(X)
+    assert_array_equal(
+        encoder.transform(y),
+        np.array([[1.,  1.,  1.,  0.,  1.,  0.,  0.]]))
+
+    encoder = UnaryEncoder(handle_greater='clip', n_values=[3, 2, 2])
+    assert_array_equal(
+        encoder.fit_transform(y),
+        np.array([[1.,  1.,  1.,  1.]]))
+
+    # Raise error if handle_greater is neither clip nor error.
+    encoder = UnaryEncoder(handle_greater='42')
+    assert_raises(ValueError, encoder.fit, y)
+
+
+def test_unary_encoder_errors():
+    n_values = np.random.randint(2, 10)
+    size = np.random.randint(1, 10)
+    delta = np.random.randint(1, 10)
+
+    encoder = UnaryEncoder(n_values)
+    X = _generate_random_features_matrix(n_values, size)
+    encoder.fit(X)
+
+    # test that an error is raised when different shape
+    larger_n_values = n_values + delta
```

Comment:
This is not clear.
It should be larger `n_features` and not larger `n_values`.
This is linked to the use of `n_values` as both `n_values` and `n_features` in `_generate_random_features_matrix`.